appservice/src/intTest/groovy/nu/yona/server/RejectBuddyTest.groovy

```diff
@@ -0,0 +1,200 @@
+    @Shared
+    def bobDunnLoginID
+    @Shared
+    def richardQuinBobBuddyURL
+    @Shared
+    def bobDunnRichardBuddyURL
+    @Shared
+    def bobDunnBuddyMessageRejectURL
+    @Shared
+    def bobDunnBuddyMessageProcessURL
+    @Shared
+    def richardQuinBuddyMessageRejectURL
+    @Shared
+    def richardQuinBuddyMessageProcessURL
+
+    def 'Add user Richard Quin'(){
+        given:
+
+        when:
+        def response = appService.addUser("""{
+				"firstName":"Richard ${timestamp}",
+				"lastName":"Quin ${timestamp}",
+				"nickName":"RQ ${timestamp}",
+				"emailAddress":"rich${timestamp}@quin.net",
+				"mobileNumber":"+${timestamp}1",
+				"devices":[
+					"Nexus 6"
+				],
+				"goals":[
+					"news"
+				]
+			}""", richardQuinPassword)
+        richardQuinURL = appService.stripQueryString(response.responseData._links.self.href)
+        richardQuinLoginID = response.responseData.vpnProfile.loginID;
+
+        then:
+        response.status == 201
+        richardQuinURL.startsWith(appServiceBaseURL + appService.USERS_PATH)
+
+        cleanup:
+        println "URL Richard: " + richardQuinURL
+    }
+
+    def 'Add user Bob Dunn'(){
+        given:
+
+        when:
+        def response = appService.addUser("""{
+				"firstName":"Bob ${timestamp}",
+				"lastName":"Dunn ${timestamp}",
+				"nickName":"BD ${timestamp}",
+				"emailAddress":"bob${timestamp}@dunn.net",
+				"mobileNumber":"+${timestamp}2",
+				"devices":[
+					"iPhone 6"
+				],
+				"goals":[
+					"gambling"
+				]
+			}""", bobDunnPassword)
+        bobDunnURL = appService.stripQueryString(response.responseData._links.self.href)
+        bobDunnLoginID = response.responseData.vpnProfile.loginID;
+
+        then:
+        response.status == 201
+        bobDunnURL.startsWith(appServiceBaseURL + appService.USERS_PATH)
+
+        cleanup:
+        println "URL Bob: " + bobDunnURL
+    }
+
+    def 'Richard requests Bob to become his buddy'(){
+        given:
+
+        when:
+        def response = appService.requestBuddy(richardQuinURL, """{
+				"_embedded":{
+					"user":{
+						"firstName":"Bob ${timestamp}",
+						"lastName":"Dun ${timestamp}n",
+						"emailAddress":"bob${timestamp}@dunn.net",
+						"mobileNumber":"+ ${timestamp}2"
+					}
+				},
+				"message":"Would you like to be my buddy?"
+			}""", richardQuinPassword)
+        richardQuinBobBuddyURL = response.responseData._links.self.href
+
+        then:
+        response.status == 201
+        response.responseData._embedded.user.firstName == "Bob ${timestamp}"
+        richardQuinBobBuddyURL.startsWith(richardQuinURL)
+
+        cleanup:
+        println "URL buddy Richard: " + richardQuinBobBuddyURL
+    }
+
+    def 'Bob checks his direct messages'(){
+        given:
+
+        when:
+        def response = appService.getDirectMessages(bobDunnURL, bobDunnPassword)
+        bobDunnBuddyMessageRejectURL = response.responseData._embedded.buddyConnectRequestMessages[0]._links.reject.href
```

Comment:
Only do this when the status is 201 and ``_links`` and ``_links.reject` are available, to prevent unreadable error messages.